sound/soc/intel/boards/sof_sdw.c

```diff
@@ -15,9 +15,32 @@
 #include "sof_sdw_common.h"
 
 unsigned long sof_sdw_quirk = SOF_RT711_JD_SRC_JD1;
+static int quirk_override = -1;
+module_param_named(quirk, quirk_override, int, 0444);
+MODULE_PARM_DESC(quirk, "Board-specific quirk override");
```

Comment:
The UVC driver also uses quirks for different not fully compliant cameras, and I remember a discussion where the driver maintainer explained, that he explicitly prohibits forcing those quirks by users because in his opinion they are pure hardware features and as such users should have no say about them. Of course, in the UVC case you can automatically apply those quirks based on vendor and device ID (maybe also on version etc.). Is there really no way to automatically detect that add-on board that you describe in the commit message?

Reply:
in my case, there's no way to detect that the platform (ICL-RVP) was reworked and no longer follows the default setup. This is not for the average user, for those we have existing DMI quirks. This is to override pre-existing quirks or set a new value if the default is not good enough in the absence of a DMI quirk

Reply:
it's the same for jack-detection quirks, they are not discoverable. You need either a DMI quirk, or pass the information manually with a kernel parameter if your platform is not handled with DMI quirks.